core/controlplane/config/templates/stack-template.json

```diff
@@ -86,40 +78,23 @@
             }
           },
           {{ end }}
-          "etcd-server-env": {
-            "files" : {
-              "/var/run/coreos/etcd-environment": {
-                "content": { "Fn::Join" : [ "", [
-                  "ETCD_INITIAL_CLUSTER='",
-                  {{range $index, $_ := $.EtcdInstances}}
-                  {{if $index}}",", {{end}}
-                  { "Fn::GetAtt" : [ "InstanceEtcd{{$index}}", "PrivateDnsName" ] },
-                  "=https://",
-                  { "Fn::GetAtt" : [ "InstanceEtcd{{$index}}", "PrivateDnsName" ] },
-                  ":2380",
-                  {{end}}
-                  "'\n"
-                ]]}
-              }
-            }
-          },
           "etcd-client-env": {
             "files" : {
               "/var/run/coreos/etcd-environment": {
                 "content": { "Fn::Join" : [ "", [
                   "ETCD_ENDPOINTS='",
-{{range $index, $_ := $.EtcdInstances}}
+                  {{range $index, $etcdInstance := $.EtcdInstances}}
                   {{if $index}}",", {{end}} "https://",
-                    { "Fn::GetAtt" : [ "InstanceEtcd{{$index}}", "PrivateDnsName" ] }, ":2379",
-{{end}}
+                  {{$etcdInstance.AdvertisedHostNameRef}}, ":2379",
+                  {{end}}
                   "'\n"
                 ]]}
               }
             }
           }
         }
       },
-      "DependsOn": ["InstanceEtcd0"]
+      "DependsOn": ["{{$.Etcd.LogicalName}}{{minus $.EtcdCount 1}}"]
```

Comment:
Currently, the controller ASG depends on the last etcd ASG. However, we can improve it further by making it depends on `(N/2)+1`th etcd ASG for earlier creation of the controller ASG.